backend/curation/api/v1/curation/collections/collection_id/datasets/dataset_id/manifest/actions.py

```diff
@@ -0,0 +1,58 @@
+from flask import Response
+
+from backend.common.utils.exceptions import MaxFileSizeExceededException
+from backend.common.utils.http_exceptions import (
+    ForbiddenHTTPException,
+    InvalidParametersHTTPException,
+    MethodNotAllowedException,
+    NotFoundHTTPException,
+    TooLargeHTTPException,
+)
+from backend.curation.api.v1.curation.collections.common import (
+    _get_collection_and_dataset,
+)
+from backend.layers.auth.user_info import UserInfo
+from backend.layers.business.exceptions import (
+    CollectionIsPublishedException,
+    CollectionNotFoundException,
+    DatasetInWrongStatusException,
+    DatasetNotFoundException,
+    InvalidURIException,
+)
+from backend.portal.api.providers import get_business_logic
+
+
+def put(collection_id: str, dataset_id: str, body: dict, token_info: dict):
+    # TODO: deduplicate from ApiCommon. We need to settle the class/module level debate before can do that
+    business_logic = get_business_logic()
+
+    collection_version, dataset_version = _get_collection_and_dataset(collection_id, dataset_id)
+
+    if not UserInfo(token_info).is_user_owner_or_allowed(collection_version.owner):
+        raise ForbiddenHTTPException()
+
+    try:
+        business_logic.ingest_dataset(
+            collection_version.version_id,
+            body,
+            None,
+            None if dataset_id is None else dataset_version.version_id,
```

Comment:
when is this none?

Reply:
I don't know. I copied this over from the other `put` implementation https://github.com/chanzuckerberg/single-cell-data-portal/blob/3652b718bfad452f730407f7e677854c5ffe1874/backend/curation/api/v1/curation/collections/collection_id/datasets/dataset_id/actions.py#L126

To me, `_get_collection_and_dataset` looks like it shouldn't return a `None` here.